libbeat/cfgfile/cfgfile.go

```diff
@@ -133,30 +134,36 @@ func Read(out interface{}, path string) error {
 // Load reads the configuration from a YAML file structure. If path is empty
 // this method reads from the configuration file specified by the '-c' command
 // line flag.
+// This function cares about the underlying fleet setting, and if beats is running with
+// the management.enabled flag, Load() will bypass reading a config file, and merely merge any overrides.
```

Comment:
> whatever was constructed by all the default configuration constructors?

Basically, yes. Now that the `UnitExpected` change in Agent is merged, I can start testing outside of the mocked environment. However, it seems like that might already be a problem with V1 when defaults are specified in a file? Not sure.

Reply:
If we want to make this cleaner somehow, the solution would be to "block" the reloaders from even starting until they get a config from the V2 controller. However, this would probably require some changes on a per-beat basis.